addons/website/static/src/snippets/s_share/000.js

```diff
@@ -59,7 +59,13 @@ const ShareWidget = publicWidget.Widget.extend({
 
         const url = encodeURIComponent(window.location.href);
         const title = encodeURIComponent(document.title);
-        const media = encodeURIComponent(document.querySelector('meta[property="og:image"]').content);
+
+        const ogImageEl = document.querySelector('meta[property="og:image"]');
+        let media;
+        // Some share links don't have a og:image. (e.g. Certification sharer)
+        if (ogImageEl) {
+            media = encodeURIComponent(ogImageEl.content);
+        }
```

Comment:
@qsm-odoo I'm not sure you saw my message above :arrow_up:

Reply:
@Guillaume-gdi 

> I've actually moved the code to have the image to share. So we only look for the image to share if the href need one to share.

That's good indeed :+1: 

> If the href mentions media, then there must be media to share in the document

I think we should be more defensive: from what I see when adding the "share" snippet somewhere, it comes with "media" in the URL for some social media (why only some... don't know...), so we should consider the snippet can be added in a page without media without bugging?

I tested 15.2, could not reproduce the issue following your steps... could you verify the bug is still there / make your steps clearer? Thank you :+1:

Reply:
> I think we should be more defensive

Indeed, you've convinced me, I'm going to make a more defensive code.

> I tested 15.2, could not reproduce the issue following your steps

Yes I can only reproduce the error from 16.0, I will move the note in my commit message about that.
![image](https://github.com/odoo/odoo/assets/78849981/17440b0f-842e-4897-85b9-e6a8e2c75c7d)
But I think that is better to solve in 15.2 because it's from this version we changed the Jquery to the vanilla JS code which cause the issue as you can see [here](https://github.com/odoo/odoo/pull/83503/commits/4a9aaa05b4abadde8fceb62de803f324344bc2c2#diff-dddc40541afc61c11513d3d14a2c2cf20f1f08788fc5b04fa24f97b2e79e95f1R53)

If you prefer, I can retarget to 16.0

Reply:
Would not have wanted you to retarget anyway :) I missed your note, strange because I read the commit message :joy: Tell me once I can have a look again :)